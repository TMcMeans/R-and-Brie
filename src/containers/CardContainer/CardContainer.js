import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Card } from '../../components/Card/Card';
import { fetchRecipes } from '../../thunks/fetchRecipes';
import { addRecipes } from '../../actions/index';
import { Error } from '../../components/Error/Error'

export class CardContainer extends Component {

  render() {
    const { label, isLoading, recipes } = this.props
    const recipeCards = recipes.map(recipe => {
      return <Card recipe={recipe} key={recipe.label} />;
    });
    if (!label) {
      return <Error />;
    } else if (isLoading) {
      return <div>Loading</div>
    } else {
      return (<div className="card-container">{recipeCards}</div>)
    }
  }
}

export const mapStateToProps = state => ({
  recipes: state.recipes,
  isLoading: state.isLoading
});

export default connect(
  mapStateToProps
)(CardContainer);

CardContainer.propTypes = {
  recipes: PropTypes.array.isRequired,
  isLoading: PropTypes.bool.isRequired
}