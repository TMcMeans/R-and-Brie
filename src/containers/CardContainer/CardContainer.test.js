import React from 'react'
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps } from './CardContainer'

describe('CardContainer', () => {
  it('should match snapshot with correct data passed in', () => {
    const wrapper = shallow(<CardContainer recipes={[{ label: 'baked brie', calories: 1144 }, { label: 'brie croissants', calories: 450 }]} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an array of recipes', () => {
      const mockState = {
        recipes: [{ label: 'baked brie', calories: 1144 }, { label: 'brie croissants', calories: 450 }],
        isLoading: false
      }
      const expected = mockState
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })


    it('should return a boolean for isLoading', () => {
      const mockState = {
        recipes: [{ label: 'baked brie', calories: 1144 }, { label: 'brie croissants', calories: 450 }],
        isLoading: false
      }
      const expected = mockState
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})
