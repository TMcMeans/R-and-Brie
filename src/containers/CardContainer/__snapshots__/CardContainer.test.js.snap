// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot with correct data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    recipes={
      Array [
        Object {
          "calories": 1144,
          "label": "baked brie",
        },
        Object {
          "calories": 450,
          "label": "brie croissants",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
